syntax = "proto3";

package flowpipe.stages.file.sink.v1;

// Configuration for FileSink stage
message FileSinkConfig {
  enum CompressionType {
    COMPRESSION_UNSPECIFIED = 0;
    COMPRESSION_NONE = 1;
    COMPRESSION_GZIP = 2;
  }

  // File path to write
  string path = 1;

  // Append to existing file (default: false)
  bool append = 2;

  // Compression type (default: none)
  CompressionType compression = 3;

  // Compression level for gzip (0-9, <=0 uses default)
  int32 compression_level = 4;
}
